packages/transactions/src/fetch.ts

```diff
@@ -0,0 +1,569 @@
+import { ClarityAbi, bytesToHex, validateHash256, with0x } from '@stacks/common';
+import { deriveDefaultUrl } from './api';
+import { estimateTransactionByteLength } from './builders';
+import { ClarityValue, NoneCV, deserializeCV, serializeCV } from './clarity';
+import { NoEstimateAvailableError } from './errors';
+import { serializePayload } from './payload';
+import { StacksTransaction, deriveNetwork } from './transaction';
+import {
+  FeeEstimateResponse,
+  FeeEstimation,
+  TxBroadcastResult,
+  TxBroadcastResultOk,
+  TxBroadcastResultRejected,
+} from './types';
+import { cvToHex, parseReadOnlyResponse } from './utils';
+
+export const HIRO_MAINNET_URL = 'https://stacks-node-api.mainnet.stacks.co';
+export const HIRO_TESTNET_URL = 'https://stacks-node-api.testnet.stacks.co';
+export const DEVNET_URL = 'http://localhost:3999';
+
+export const BROADCAST_PATH = '/v2/transactions';
+export const TRANSFER_FEE_ESTIMATE_PATH = '/v2/fees/transfer';
+export const TRANSACTION_FEE_ESTIMATE_PATH = '/v2/fees/transaction';
+export const ACCOUNT_PATH = '/v2/accounts';
+export const CONTRACT_ABI_PATH = '/v2/contracts/interface';
+export const READONLY_FUNCTION_CALL_PATH = '/v2/contracts/call-read';
+export const MAP_ENTRY_PATH = '/v2/map_entry';
```

Comment:
> Should StacksApi be StacksNode or StacksNodeApi

With almost all the endpoints being for the stacks node RPC, I'd be in favor of calling it something other than "API" because that ends up confusing a lot of people

Reply:
I'll rename to `StacksNodeApi`. I'll also add methods for calling mempool-nonce estimates (via API) in a PR soon. I'll keep the behavior to fallback to node fetches if API fetches don't work.